src/cnd/cnd.ts

```diff
@@ -44,6 +44,50 @@ export interface SwapRequest {
   peer: Peer;
 }
 
+export interface CoreRequestBody {
+  // to_public_key: string; // This is exchanged via libp2p
+  // secret_hash: string; // Generated by cnd
+  // alpha_ledger: string; // Ledger name is in the route
+  // beta_ledger: string; // Ledger name is in the route
+  alpha_amount: string; // Asset name is in the route
```

Comment:
At the expense of having to define more interfaces: should we consider _not_ having alpha/beta terminology in the API but just say:

`satoshis` for example?

Or maybe a combination of both worlds? Although that would be quite weird.

The motivation would be that we don't need to try and maintain a terminology across all ledgers and protocols and fit it in. A good example is `final_cltv_expiry`, how is the user going to know what should go in this field?

Maybe the current approach is fine, I just wanted to highlight that with different routes per combination, we have the freedom to do something different.

Reply:
> At the expense of having to define more interfaces: should we consider _not_ having alpha/beta terminology in the API but just say:
> 
> `satoshis` for example?
> 
> Or maybe a combination of both worlds? Although that would be quite weird.
> 
> The motivation would be that we don't need to try and maintain a terminology across all ledgers and protocols and fit it in. A good example is `final_cltv_expiry`, how is the user going to know what should go in this field?

Hum, sorry, what `final_cltv_expiry` field? There is no such field defined in this proposal.

> 
> Maybe the current approach is fine, I just wanted to highlight that with different routes per combination, we have the freedom to do something different.

True. I went for homogeneity across the route but it may not be justifiable.
The issues with `satoshis` is: What would we propose for ERC-20 tokens?
What if we are dealing with WBTC? it might create some confusions I belive.

Reply:
> What would we propose for ERC-20 tokens? What if we are dealing with WBTC? it might create some confusions I believe.

Very good questions!
How much time do you recon is it worth to invest now here instead of making breaking changes later?

Reply:
> > What would we propose for ERC-20 tokens? What if we are dealing with WBTC? it might create some confusions I believe.
> 
> Very good questions!
> How much time do you recon is it worth to invest now here instead of making breaking changes later?

Well to avoid breaking changes I'd be keen to use `alpha_amount` terminology.

Reply:
Okay, let's roll with it and see where we end up :)